packages/babel-helper-transform-fixture-test-runner/src/index.js

```diff
@@ -130,6 +130,156 @@ function wrapPackagesArray(type, names, optionsDir) {
+                        isByRegenerator(parent.id),
+                    ),
+                ),
+            ),
+        )
+      ) {
+        // regeneratorRuntime.mark(function _callee3() {
+        //   var bar, _bar2; // <-- Those identifiers
+        return true;
+      } else if (
+        parentIs(
+          node,
+          parent =>
+            parent.type === "VariableDeclarator" &&
+            parentIs(
+              parent,
+              parent =>
+                parent.type === "VariableDeclaration" &&
+                parentIs(
+                  parent,
+                  parent =>
+                    parent.type === "BlockStatement" &&
+                    parent.body.length === 2 &&
+                    parent.body[1].type === "ReturnStatement" &&
+                    parent.body[1].argument.type === "CallExpression" &&
+                    parent.body[1].argument.callee.type ===
+                      "MemberExpression" &&
+                    parent.body[1].argument.callee.property.type ===
+                      "Identifier" &&
+                    parent.body[1].argument.callee.property.name === "wrap",
+                ),
+            ),
+        )
+      ) {
+        // function foo() {
+        //   var _len, // <-- Those identifiers
+        //     items,
+        //     _key,
+        //     _args = arguments;
+        //   return regeneratorRuntime.wrap(function foo$(_context) {
+        return true;
+      } else if (
+        parentIs(
+          node,
+          parent =>
+            parent.type === "CallExpression" &&
+            parent.callee.type === "MemberExpression" &&
+            parent.callee.property.type === "Identifier" &&
+            parent.callee.property.name === "mark",
+        )
+      ) {
+        // regeneratorRuntime.mark(foo); // foo
+        return true;
+      }
+    } else if (node.type === "MemberExpression") {
+      // _context.next
+      return isByRegenerator(node.object);
+    } else if (node.type === "CallExpression") {
+      return isByRegenerator(node.callee);
+    } else if (node.type === "AssignmentExpression") {
+      // _context.next = 4;
+      return isByRegenerator(node.left);
+    } else if (node.type === "NumericLiteral") {
+      // _context.next = 4; // <-- The 4
+      return parentIs(
+        node,
+        parent =>
+          parent.type === "AssignmentExpression" &&
+          isByRegenerator(parent.left),
+      );
+    }
+    return false;
+  };
+  const hidePrivateProperties = (key, val) => {
+    // Hides properties like _shadowedFunctionLiteral,
+    // which makes the AST circular
+    if (key[0] === "_") return "[Private]";
+    return val;
+  };
+  babel.types.traverseFast(ast, node => {
+    registerChildren(node);
+    if (isByRegenerator(node)) return;
+    if (nodes.has(node)) {
+      throw new Error(
+        "Do not reuse nodes. Use `t.clone` or `t.cloneDeep` to copy them.\n" +
```

Comment:
(Q: when to _not_ use cloneDeep?)

Reply:
When you are working with a literal, which doesn't have children

Reply:
But if you are working with a literal, cloneDeep just won't find additional objects to recurse and so it won't recurse. Any reason why we can't just provide a single interface to reduce errors? Would the performance impact of the extra typeof check be too great?

Reply:
We can make `clone` behave as `cloneDeep` and special-case literals to avoid looping over their properties.